pkg/collectors/pod.go

```diff
@@ -429,35 +418,84 @@ func (pc *podCollector) collectPod(ch chan<- prometheus.Metric, p v1.Pod) {
 		addGauge(descPodCompletionTime, lastFinishTime)
 	}
 
-	for _, c := range p.Spec.Containers {
-		req := c.Resources.Requests
-		lim := c.Resources.Limits
+	if !pc.opts.DisablePodNoneGenericResourceMetrics {
+		for _, c := range p.Spec.Containers {
+			req := c.Resources.Requests
+			lim := c.Resources.Limits
 
-		if cpu, ok := req[v1.ResourceCPU]; ok {
-			addGauge(descPodContainerResourceRequestsCpuCores, float64(cpu.MilliValue())/1000,
-				c.Name, nodeName)
-		}
-		if mem, ok := req[v1.ResourceMemory]; ok {
-			addGauge(descPodContainerResourceRequestsMemoryBytes, float64(mem.Value()),
-				c.Name, nodeName)
-		}
+			if cpu, ok := req[v1.ResourceCPU]; ok {
+				addGauge(descPodContainerResourceRequestsCpuCores, float64(cpu.MilliValue())/1000,
+					c.Name, nodeName)
+			}
+			if mem, ok := req[v1.ResourceMemory]; ok {
+				addGauge(descPodContainerResourceRequestsMemoryBytes, float64(mem.Value()),
+					c.Name, nodeName)
+			}
 
-		if gpu, ok := req[v1.ResourceNvidiaGPU]; ok {
-			addGauge(descPodContainerResourceRequestsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
-		}
+			if gpu, ok := req[v1.ResourceNvidiaGPU]; ok {
+				addGauge(descPodContainerResourceRequestsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
+			}
+
+			if cpu, ok := lim[v1.ResourceCPU]; ok {
+				addGauge(descPodContainerResourceLimitsCpuCores, float64(cpu.MilliValue())/1000,
+					c.Name, nodeName)
+			}
+
+			if mem, ok := lim[v1.ResourceMemory]; ok {
+				addGauge(descPodContainerResourceLimitsMemoryBytes, float64(mem.Value()),
+					c.Name, nodeName)
+			}
 
-		if cpu, ok := lim[v1.ResourceCPU]; ok {
-			addGauge(descPodContainerResourceLimitsCpuCores, float64(cpu.MilliValue())/1000,
-				c.Name, nodeName)
+			if gpu, ok := lim[v1.ResourceNvidiaGPU]; ok {
+				addGauge(descPodContainerResourceLimitsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
+			}
 		}
+	}
+
+	for _, c := range p.Spec.Containers {
+		req := c.Resources.Requests
+		lim := c.Resources.Limits
 
-		if mem, ok := lim[v1.ResourceMemory]; ok {
-			addGauge(descPodContainerResourceLimitsMemoryBytes, float64(mem.Value()),
-				c.Name, nodeName)
+		for resourceName, val := range req {
+			switch resourceName {
+			case v1.ResourceCPU:
+				addGauge(descPodContainerResourceRequests, float64(val.MilliValue())/1000,
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "core")
+			case v1.ResourceStorage:
+				fallthrough
+			case v1.ResourceEphemeralStorage:
```

Comment:
this won't label `hugepages-*` properly as bytes.  any resource name that starts with `hugepages-` is measured in bytes.

Reply:
@derekwaynecarr 
> ephemeral storage is bytes as well.

This has been the truth as when the resource name is `ResourceEphemeralStorage` and `ResourceStorage`, it will `fallthrough` to `ResourceMemory`.

> this won't label hugepages-* properly as bytes. any resource name that starts with hugepages- is measured in bytes.

I have actually not aware about this. Will do.